sklearn/model_selection/_split.py

```diff
@@ -2787,6 +2789,14 @@ def train_test_split(
 
         train, test = next(cv.split(X=arrays[0], y=stratify))
 
+    xp, is_array_api_compliant = get_namespace(arrays[0])
+    if is_array_api_compliant:
+        # For indexing, the data and index arrays need to be of the same type
+        # and on the same device.
+        _device = array_api_device(arrays[0])
+        train = xp.asarray(train, device=_device)
+        test = xp.asarray(test, device=_device)
```

Comment:
In a previous PR someone commented on and we went with `device_` as the solution to not shadowing `device`, instead of using a leading `_` (which often means "private" in Python code). The consistency hobgoblin in me prefers the trailing `_` for the sole reason that it is consistent.
```suggestion
        device_ = array_api_device(arrays[0])
        train = xp.asarray(train, device=device_)
        test = xp.asarray(test, device=device_)
```